### PR TITLE
tarantool 2.5.2.0

### PR DESCRIPTION
**Pull request by avtikhon**
Set stable release to 2.5.2.0 version.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by avtikhon** [comment]
Hi,
I don't actually see any real issue, is it possible to restart testing to recheck it and continue with pull request ?

**Comment by gromgit** [comment]
https://github.com/Homebrew/homebrew-core/pull/59203/checks?check_run_id=949689231#step:6:4187
```
/tmp/tarantool-20200805-69958-vjqsuw/tarantool-2.4.2.1/src/lua/init.c:478:2: error: use of undeclared identifier 'rl_catch_signals'
        rl_catch_signals = 0;
        ^
/tmp/tarantool-20200805-69958-vjqsuw/tarantool-2.4.2.1/src/lua/init.c:479:2: error: use of undeclared identifier 'rl_catch_sigwinch'
        rl_catch_sigwinch = 0;
        ^
2 errors generated.
```
Those are `readline` extern symbols. Odd that only 10.13 build fails.

> is it possible to restart testing to recheck

`brew update`, rebase your PR branch against `origin/master` then force-push.

**Comment by avtikhon** [comment]
I've rebased PR branch against origin/master then force-pushed, but seems more issues appeared.

**Comment by Totktonada** [comment]
It seems, tarantool fails to build when GNU readline from brew is installed into the system: the ifdef'ed code under HAVE_GNU_READLINE macro is enabled, but the directory with the GNU readline header files is not passed with `-I`.

The problem is on our side: https://github.com/tarantool/tarantool/issues/5322

File: Formula/tarantool.rb [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+  sha256 "f64b8ef772d42017bd938770912b4a1727ce6302087efcd5c7faac9e425d9ac4"
   license "BSD-2-Clause"
-  head "https://github.com/tarantool/tarantool.git", branch: "2.3", shallow: false
+  head "https://github.com/tarantool/tarantool.git", branch: "2.5", shallow: false
 
   bottle do
     cellar :any
     sha256 "e013978027f78ed680b09fd07ddc19045852ea6eb158a98af580826dd363cbb9" => :catalina
     sha256 "8cd34bf1c2129621a127a59406037a18a2ebe2c3d11d08d8397766568d2057a4" => :mojave
-    sha256 "a66ae22addde4398e8d5300799c839ce404bae2925f2f94a0b95a0b40af7ac81" => :high_sierra
```
Comment (Totktonada) [inline Formula/tarantool.rb:13]: Looks as the bad idea.

**Comment by Totktonada** [comment]
Regarding the fail on 10.13:

I able to reproduce and investigate it with the following command:

```sh
brew install --verbose --keep-tmp --debug tarantool
```

As I see, under brew the compiler reports `/usr/local/opt` paths as default:

```
#include <...> search starts here:
 /usr/local/opt/icu4c/include
 /usr/local/opt/openssl@1.1/include
 /usr/local/opt/readline/include
 /usr/local/include
 /usr/include/libxml2
 /System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
 /Library/Developer/CommandLineTools/usr/lib/clang/10.0.0/include
 /Library/Developer/CommandLineTools/usr/include
 /usr/include
 /System/Library/Frameworks (framework directory)
 /Library/Frameworks (framework directory)
End of search list.
```

<details>
<summary>The paths from direct invoke cc without shims (under cut):</summary>

```
#include <...> search starts here:
 /Library/Developer/CommandLineTools/usr/lib/clang/10.0.0/include
 /Library/Developer/CommandLineTools/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks (framework directory)
End of search list.
```
</details>

Those paths are not passed using `-I` to the compiler: they're excluded by cmake.

We append curses include paths into the readline ones this way:

```cmake
set(READLINE_INCLUDE_DIRS ${READLINE_INCLUDE_DIRS} ${CURSES_INCLUDE_DIRS})
```

The initial READLINE_INCLUDE_DIRS is `/usr/local/opt/readline/include` (excluded by cmake) and `CURSES_INCLUDE_DIRS` is `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include` (` /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/` on my machine) (not excluded, because it is not listed in the default compiler search paths).

So `include_directories(${READLINE_INCLUDE_DIRS})` adds the SDK's headers, where readline.h does not have the following externs:

```c
$ grep rl_catch_sig /usr/local/opt/readline/include/readline/readline.h      
extern int rl_catch_signals;
extern int rl_catch_sigwinch;
```

I don't know why dependencies headers are included into the default search paths (for convenience?), but okay. I don't know why SDK's headers are not included into the default paths (but okay). Anyway, we can just set ncurses header and library explicitly in the formula:

```ruby
    args << "-DCURSES_NEED_NCURSES=TRUE"
    args << "-DCURSES_NCURSES_INCLUDE_PATH=/usr/include"
    args << "-DCURSES_NCURSES_LIBRARY=/usr/lib/libncurses.dylib"
```

After that I found that the similar problem appears for iconv:

```ruby
    args << "-DICONV_INCLUDE_DIR=/usr/include"
```

After this the `brew install tarantool` command passes (for the 2.5.2.0 version). Maybe there is a better way to overcome the problem, I don't know.

Side notes:

1. I see that `uses_from_macos "ncurses"` clause is used in some formulae. The tarantool formula does not have it. Should it?
2. We use bundled libcurl, so `-DCURL_INCLUDE_DIR` and `-DCURL_LIBRARY` have no effect (see the reasoning behind this [here](https://github.com/tarantool/tarantool/issues/4318), it is mainly for CentOS 7). Homebrew maintainers may decide to drop those flags, or keep them and pass `-DENABLE_BUNDLED_LIBCURL=OFF`.

File: Formula/tarantool.rb [hunk 2]
```diff
@@ -17,9 +17,9 @@ class Tarantool < Formula
   depends_on "automake" => :build
   depends_on "cmake" => :build
   depends_on "libtool" => :build
+  depends_on "readline" => :build
```
[the remark this answers is not in the document]
Comment (avtikhon) [reply Formula/tarantool.rb]: Right, reverted.

File: Formula/tarantool.rb (hunk 1 above)
Comment (avtikhon) [reply Formula/tarantool.rb:13]: Right, restored.

File: Formula/tarantool.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 class Tarantool < Formula
   desc "In-memory database and Lua application server"
   homepage "https://tarantool.org/"
-  url "https://hb.bizmrg.com/tarantool.2.3.src/tarantool-2.3.2.81.tar.gz"
-  sha256 "312ee4b68a0834e01a84df0fa1eb34dd7484a5590f68e1184599ed4dd5c575a4"
+  url "https://hb.bizmrg.com/tarantool_repo/sources/2.5/tarantool-2.5.2.1.tar.gz"
```
[the remark this answers is not in the document]
Comment (avtikhon) [reply Formula/tarantool.rb]: Changed.

**Comment by avtikhon** [comment]
> Regarding the fail on 10.13:
> 
> I able to reproduce and investigate it with the following command:
> 
> ```shell
> brew install --verbose --keep-tmp --debug tarantool
> ```
> 
> As I see, under brew the compiler reports `/usr/local/opt` paths as default:
> 
> ```
> #include <...> search starts here:
>  /usr/local/opt/icu4c/include
>  /usr/local/opt/openssl@1.1/include
>  /usr/local/opt/readline/include
>  /usr/local/include
>  /usr/include/libxml2
>  /System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
>  /Library/Developer/CommandLineTools/usr/lib/clang/10.0.0/include
>  /Library/Developer/CommandLineTools/usr/include
>  /usr/include
>  /System/Library/Frameworks (framework directory)
>  /Library/Frameworks (framework directory)
> End of search list.
> ```
> 
> The paths from direct invoke cc without shims (under cut):
> Those paths are not passed using `-I` to the compiler: they're excluded by cmake.
> 
> We append curses include paths into the readline ones this way:
> 
> ```cmake
> set(READLINE_INCLUDE_DIRS ${READLINE_INCLUDE_DIRS} ${CURSES_INCLUDE_DIRS})
> ```
> 
> The initial READLINE_INCLUDE_DIRS is `/usr/local/opt/readline/include` (excluded by cmake) and `CURSES_INCLUDE_DIRS` is `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include` (` /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/` on my machine) (not excluded, because it is not listed in the default compiler search paths).
> 
> So `include_directories(${READLINE_INCLUDE_DIRS})` adds the SDK's headers, where readline.h does not have the following externs:
> 
> ```c
> $ grep rl_catch_sig /usr/local/opt/readline/include/readline/readline.h      
> extern int rl_catch_signals;
> extern int rl_catch_sigwinch;
> ```
> 
> I don't know why dependencies headers are included into the default search paths (for convenience?), but okay. I don't know why SDK's headers are not included into the default paths (but okay). Anyway, we can just set ncurses header and library explicitly in the formula:
> 
> ```ruby
>     args << "-DCURSES_NEED_NCURSES=TRUE"
>     args << "-DCURSES_NCURSES_INCLUDE_PATH=/usr/include"
>     args << "-DCURSES_NCURSES_LIBRARY=/usr/lib/libncurses.dylib"
> ```
> 
> After that I found that the similar problem appears for iconv:
> 
> ```ruby
>     args << "-DICONV_INCLUDE_DIR=/usr/include"
> ```
> 
> After this the `brew install tarantool` command passes (for the 2.5.2.0 version). Maybe there is a better way to overcome the problem, I don't know.
> 
> Side notes:
> 
> 1. I see that `uses_from_macos "ncurses"` clause is used in some formulae. The tarantool formula does not have it. Should it?
> 2. We use bundled libcurl, so `-DCURL_INCLUDE_DIR` and `-DCURL_LIBRARY` have no effect (see the reasoning behind this [here](https://github.com/tarantool/tarantool/issues/4318), it is mainly for CentOS 7). Homebrew maintainers may decide to drop those flags, or keep them and pass `-DENABLE_BUNDLED_LIBCURL=OFF`.

Thanks a lot for your investigation and complete fix, I've used it in the patch and it really fixed the issue.

**Comment by Totktonada** [comment]
@SeekingMeaning The CI succeeds now. What our next actions?

I would kindly ask maintainers to review the approach to get rid of SDK's include directories (it is described above).

**Comment by Totktonada** [comment]
> 2. We use bundled libcurl, so `-DCURL_INCLUDE_DIR` and `-DCURL_LIBRARY` have no effect (see the reasoning behind this [here](https://github.com/tarantool/tarantool/issues/4318), it is mainly for CentOS 7). Homebrew maintainers may decide to drop those flags, or keep them and pass `-DENABLE_BUNDLED_LIBCURL=OFF`.

Forgot to mention. The following dependencies are only necessary for building the bundled libcurl:

```ruby
  depends_on "autoconf" => :build
  depends_on "automake" => :build
  depends_on "libtool" => :build
```

If the decidion will be to link against the system libcurl, those dependencies may be removed.

**Comment by SMillerDev** [comment]
Can it link against the homebrew libcurl instead?

**Comment by Totktonada** [comment]
Yep, we bundle unchanged libcurl, just to fix its version. I don't mind, those decisions are always up to distribution maintainers.

@avtikhon Can you add `-DENABLE_BUNDLED_LIBCURL=OFF`?

**Comment by SMillerDev** [comment]
and make it depend on `curl` then

File: Formula/tarantool.rb [hunk 4]
```diff
@@ -39,8 +37,13 @@ def install
     args << "-DENABLE_DIST=ON"
     args << "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}"
     args << "-DREADLINE_ROOT=#{Formula["readline"].opt_prefix}"
+    args << "-DENABLE_BUNDLED_LIBCURL=OFF"
     args << "-DCURL_INCLUDE_DIR=#{sdk}/usr/include"
     args << "-DCURL_LIBRARY=/usr/lib/libcurl.dylib"
```
Comment (SMillerDev) [inline Formula/tarantool.rb:42]: I don't think this is correct then. I'd expect it to be 
```suggestion
    args << "-DCURL_INCLUDE_DIR=#{Formula["curl"].opt_include}"
    args << "-DCURL_LIBRARY=#{Formula["curl"].opt_lib}/libcurl.dylib"
```

Comment (avtikhon) [reply Formula/tarantool.rb]: Seems, that the new issue occurred on OSX 13 build with it:
```
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201102-74692-yws255.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/tarantool.rb --verbose
==> /usr/local/Cellar/tarantool/2.5.2.0/bin/tarantool /private/tmp/tarantool-test-20201102-74693-6rwmzr/test.lua
dyld: Library not loaded: /usr/local/opt/curl/lib/libcurl.4.dylib
  Referenced from: /usr/local/Cellar/tarantool/2.5.2.0/bin/tarantool
  Reason: Incompatible library version: tarantool requires version 12.0.0 or later, but libcurl.4.dylib provides version 7.0.0
Error: tarantool: failed
An exception occurred within a child process:
  BuildError: Failed executing: /usr/local/Cellar/tarantool/2.5.2.0/bin/tarantool /private/tmp/tarantool-test-20201102-74693-6rwmzr/test.lua
```

**Comment by Totktonada** [comment]
> Can it link against the homebrew libcurl instead?

Ouch, I misread it as 'against system libcurl', because we was [asked](https://github.com/Homebrew/homebrew-core/pull/18532) for this before. I don't mind any decision, though.

File: Formula/tarantool.rb (hunk 4 above)
Comment (Totktonada) [reply Formula/tarantool.rb]: Alexander, it is not only build-time dependency, but runtime as well (we'll need the library on the target system). Let's remove `:build`.

**Comment by SMillerDev** [comment]
That would also be fine, but I doubt that version is up to date. If that works you can add `uses_from_macos "curl"`

File: Formula/tarantool.rb [hunk 5]
```diff
@@ -39,8 +36,13 @@ def install
     args << "-DENABLE_DIST=ON"
     args << "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}"
     args << "-DREADLINE_ROOT=#{Formula["readline"].opt_prefix}"
-    args << "-DCURL_INCLUDE_DIR=#{sdk}/usr/include"
-    args << "-DCURL_LIBRARY=/usr/lib/libcurl.dylib"
+    args << "-DENABLE_BUNDLED_LIBCURL=OFF"
+    args << "-DCURL_INCLUDE_DIR=#{Formula["curl"].opt_include}"
+    args << "-DCURL_LIBRARY=#{Formula["curl"].opt_lib}/libcurl.dylib"
```
Comment (SMillerDev) [inline Formula/tarantool.rb:41]: This should be the system path then

File: Formula/tarantool.rb [hunk 6]
```diff
@@ -39,8 +36,13 @@ def install
     args << "-DENABLE_DIST=ON"
     args << "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}"
     args << "-DREADLINE_ROOT=#{Formula["readline"].opt_prefix}"
-    args << "-DCURL_INCLUDE_DIR=#{sdk}/usr/include"
+    args << "-DENABLE_BUNDLED_LIBCURL=OFF"
+    args << "-DCURL_INCLUDE_DIR=/usr/include"
```
Comment (SMillerDev) [inline Formula/tarantool.rb:40]: I believe this, curses and iconv include should still be 
```suggestion
    args << "-DCURL_INCLUDE_DIR=#{sdk}/usr/include"
```

Comment (avtikhon) [reply Formula/tarantool.rb]: Thanks, for the suggestions, I've made them all.

File: Formula/tarantool.rb (hunk 4 above)
Comment (avtikhon) [reply Formula/tarantool.rb:43]: Ok.

File: Formula/tarantool.rb [hunk 7]
```diff
@@ -39,8 +38,13 @@ def install
     args << "-DENABLE_DIST=ON"
     args << "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}"
     args << "-DREADLINE_ROOT=#{Formula["readline"].opt_prefix}"
+    args << "-DENABLE_BUNDLED_LIBCURL=OFF"
     args << "-DCURL_INCLUDE_DIR=#{sdk}/usr/include"
     args << "-DCURL_LIBRARY=/usr/lib/libcurl.dylib"
+    args << "-DCURSES_NEED_NCURSES=TRUE"
+    args << "-DCURSES_NCURSES_INCLUDE_PATH=/usr/include"
+    args << "-DCURSES_NCURSES_LIBRARY=/usr/lib/libncurses.dylib"
+    args << "-DICONV_INCLUDE_DIR=/usr/include"
```
Comment (SMillerDev) [inline Formula/tarantool.rb:47]: I don't know what the library setting is for, but that file doesn't exist on my system

```suggestion
    args << "-DCURSES_NCURSES_INCLUDE_PATH=#{sdk}/usr/include"
    args << "-DCURSES_NCURSES_LIBRARY=/usr/lib/libncurses.dylib"
    args << "-DICONV_INCLUDE_DIR=#{sdk}/usr/include"
```

**Comment by Totktonada** [comment]
> That would also be fine, but I doubt that version is up to date. If that works you can add `uses_from_macos "curl"`

And the same for ncurses?

File: Formula/tarantool.rb [hunk 8]
```diff
@@ -13,14 +13,13 @@ class Tarantool < Formula
...
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
   depends_on "cmake" => :build
-  depends_on "libtool" => :build
   depends_on "icu4c"
   depends_on "openssl@1.1"
   depends_on "readline"
 
+  uses_from_macos "curl"
+
```
Comment (SMillerDev) [inline Formula/tarantool.rb:22]:
```suggestion
  uses_from_macos "curl"
  uses_from_macos "ncurses"

```

File: Formula/tarantool.rb (hunk 7 above)
Comment (Totktonada) [reply Formula/tarantool.rb:48]: The goal is to get rid of the SDK paths to don't hit to readline.h from it (more details are above). Hm. Whether the build passes on your system after changing the definitions this way?

Comment (SMillerDev) [reply Formula/tarantool.rb:48]: Seeing how Apple is moving away from putting anything in a UNIX standard location like this, I think we should really try and fix the readline issue rather then masking it.

File: Formula/tarantool.rb (hunk 8 above)
Comment (avtikhon) [reply Formula/tarantool.rb]: Done.

File: Formula/tarantool.rb (hunk 7 above)
Comment (avtikhon) [reply Formula/tarantool.rb]: Done.

Comment (Totktonada) [reply Formula/tarantool.rb]: Your point is valid too. Sadly, the problem looks deeper than just a broken logic in our cmake scripts. If you're curious, I can try to formulate my thoughts around it. Maybe I miss some widespread approach for libraries search / compiling against them in a complex environment with several library collections of different kinds and origins.

Should cmake ignore libraries in the SDK directories if the SDK is not in the default compiler include paths? If so, why cmake sets -isysroot?

**Review by SMillerDev — APPROVED** [review]
Thanks @avtikhon  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**